x-pack/scripts/functional_tests.js

```diff
@@ -45,6 +45,8 @@ const onlyNotInCoverageTests = [
   require.resolve('../test/detection_engine_api_integration/basic/config.ts'),
   require.resolve('../test/lists_api_integration/security_and_spaces/config.ts'),
   require.resolve('../test/plugin_api_integration/config.ts'),
+  require.resolve('.../test/rule_registry/security_and_spaces/tests/config_basic.ts'),
+  require.resolve('.../test/rule_registry/security_and_spaces/tests/config_trial.ts'),
```

Comment:
It would be a good idea to add a `spaces_only` versions of these integration tests. Given that you used `security_and_spaces` here, I imagine that was the intent at some point. This PR is already huge, perhaps in a follow-on PR?